src/form-elements/FormElementCalculation.tsx

```diff
@@ -84,6 +84,28 @@ function FormElementCalculation({ element, onChange, value }: Props) {
               // NaN is accounted for is the calculation
               // so we can return that from here
               if (typeof elementValue === 'string') {
+                // The string could be an iso date string, or a string
+                // resembling a date. We need to parse the value as an ISO string
+                // and as a date in the format below to cover all calculation checks
+                // with 'date', 'datetime' and 'time' elements. If the string is not
+                // one of these, then we want to parse it as a float.
+
+                // Date-fns has a parseIso function, but it'll interpret '10', '11', etc.
+                // as an iso date and cause issues with these string numbers. To combat this problem,
+                // using the parse function with the ISO format will bypass this.
+                const parsedIsoDate = parse(
+                  elementValue,
+                  "yyyy-MM-dd'T'HH:mm:ss.SSSX",
```

Comment:
Prob better to use https://date-fns.org/v2.30.0/docs/parseISO here instead

Reply:
The comment above is why I couldn't use that function, I had that initially but led to the issue where "10", "11" and other numbers were being parsed as ISO strings and resulting in these results being parsed as a date